rule/expr.go

```diff
@@ -173,7 +174,7 @@ func isScalar(e bzl.Expr) bool {
 	}
 }
 
-func dictEntryKeyValue(e bzl.Expr) (string, *bzl.ListExpr, error) {
+func dictEntryKeyValue(e bzl.Expr) (string, bzl.Expr, error) {
```

Comment:
It's fine to rename this function (maybe `selectDictEntryKeyValue`), but its semantics shouldn't change. It's matching entries in a dict passed to a `select`, and it needs to keep doing that.